### ISSUE TITLE
Live Regions on iOS

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#45968]
Today, setting `liveRegion` to true marks an iOS accessibility node with [`UIAccessibilityTraitUpdatesFrequently`](https://developer.apple.com/documentation/uikit/uiaccessibilitytraitupdatesfrequently?language=objc), which doesn't actually announce anything.

We _could_ make the experience a bit closer to Android by sending an explicit announcement when a live region actually updates (we already have the information we'd need to do that in the bridge).  It's not clear to me whether this would be a good idea or not though - unlike regular live region announcemnts (which in Flutter are always "polite"), it's not clear that announce would allow for interruptions from VoiceOver.

If we think this is the wrong way to go, we should probably provide better guidance than we do in the documentation beyond saying "this works in Android but not in iOS" - I would find it helpful to know what kind of approaches I should or should not consider taking (e.g., why or whether I should avoid doing `SemanticsService.announce` on iOS, or perhaps just why Flutter doesn't do that for me automatically on iOS).

/cc @goderbauer @jonahwilliams @darrenaustin @vick08 @caseyburkhardt @cbracken

**Comment by vick08** [comment]
It seems as though the "UpdatesFrequently" accessibility trait may be a closer fit here rather than posting an "announcement" notification.
Tangentially, we should probably support announcements as well even though they are generally of a disruptive nature. TO mitigate the above, iOS does post a notification when an announcement is finished speaking.

**Comment by caseyburkhardt** [comment]
@vick08 The docs for `UIAccessibilityTraitUpdatesFrequently` mentions it suppresses update notifications, and it's intended to be used where the UI updates more frequently than would be desirable for Voice Over to announce repeatedly.

To me, It looks like this is the exact opposite of what one would expect for a live region, so I think @dnfield is indeed onto something here.

**Comment by vick08** [comment]
Thx @caseyburkhardt for the clarification and for pointing out my oversight. In factt it seems that UpdatesFrequently is already used and is not announcing anything ′probably per above′.
It doesn't look like announce notifications for iOS have PriorityKey, like they do on Mac OS. So, looks like "announcement" notification is our only choice?

**Comment by caseyburkhardt** [comment]
@j-sid could probably confirm that or provide another recommendation.

**Comment by j-sid** [comment]
I am surprised that `UIAccessibilityTraitUpdatesFrequently` is not announcing anything here, elements with `UIAccessibilityTraitUpdatesFrequently` are supposed to announce themselves normally when focussed and if the focus remains for > `5 seconds` (on iPhoneX) the values must be announced again, this will repeat every `5 sec`. If this is not happening here the element might be missing some other a11y attributes.

**Comment by dnfield** [comment]
@j-sid there's a bit of a mistmatch in concepts here.

A live region on Android triggers "polite" announcements even if the element does not have focus.

There's no corresponding concept on iOS.  UpdatesFrequently is only for elements with Focus, and just makes sure that the announcement does not happen _too frequently_ (rather than making sure the announcement happens even if the element lacks focus).  From what I can tell, Safari does support some sort of live region, but only as a "rude" announcement (i.e. it will interrupt other, potentially more important, system announcements).

**Comment by cookiecrook** [comment]
If you’re attempting to convert web live regions into native UI, you’ll need to monitor for changes and fire relevant `UIAccessibilityAnnouncementNotification` announcement.

**Comment by cookiecrook** [comment]
Or, if you’ve found a WebKit bug, please file it with a reduced HTML test case to https://webkit.org/new-ax-bug thanks!

**Comment by dnfield** [comment]
@cookiecrook that's basically what I'm proposing here.

I wouldn't say there's a bug in WebKit, more a limitation around announcements on iOS.  There is no way for a developer to provide a non-interrupting announcement, and if you do provide an announcement you may interrupt a more important one from the OS.

**Comment by vick08** [comment]
@dnfield it does look as though we can resort to relying on accessibility speech attributes that can be posted along with the announcement https://developer.apple.com/documentation/foundation/nsattributedstring/key/2865770-accessibilityspeechqueueannounce  The "yes" and "no" values will decide whether the speech will queue or disrupt. This should accomplish what we want for live regions. WDYT? announcement. For example, '

**Comment by dnfield** [comment]
Ahh that's perfect, with that we should be able to have parity with Android

**Comment by j-sid** [comment]
If we are going this route then we must not have `UIAccessibilityTraitUpdatesFrequently` anymore for the live region, since iOS specifically opted for this trait to deal with elements with focus and (since iOS is a native platform) left it to the devs to post announcements when appropriate. Also note that `accessibilitySpeechQueueAnnouncement ` is `iOS 11` and above only.

**Comment by dnfield** [comment]
I have a patch prepared for the engine that enables this for iOS. Will also have to update docs once that lands in the framework.

**Comment by dnfield** [comment]
Ah. The docs are in the engine as well. The only thing relevant to the framework is a deprecated method which I do not intend to update documentation on.
